src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/AuthenticatorTemplate.cs

```diff
@@ -38,7 +38,9 @@ internal class AuthenticatorTemplate
     {
         private const string AuthorizeEndpointTemplate = "https://{host}/{tenant}/oauth2/authorize";
         private const string DeviceCodeEndpointTemplate = "https://{host}/{tenant}/oauth2/devicecode";
-        private const string MetadataTemplate = "{\"Host\":\"{host}\", \"Authority\":\"https://{host}/{tenant}/\", \"InstanceDiscoveryEndpoint\":\"https://{host}/common/discovery/instance\", \"DeviceCodeEndpoint\":\"" + DeviceCodeEndpointTemplate + "\", \"AuthorizeEndpoint\":\"" + AuthorizeEndpointTemplate + "\", \"TokenEndpoint\":\"https://{host}/{tenant}/oauth2/token\", \"UserRealmEndpoint\":\"https://{host}/common/UserRealm\"}";
+       
+        private const string MetadataTemplate = "{\"Host\":\"{host}\", \"Authority\":\"https://{host}/{tenant}/\", \"InstanceDiscoveryEndpoint\":\"https://{host}/common/discovery/instance\", \"DeviceCodeEndpoint\":\"" + DeviceCodeEndpointTemplate + "\", \"AuthorizeEndpoint\":\"" + AuthorizeEndpointTemplate + "\", \"TokenEndpoint\":\"https://{host}/{tenant}/oauth2/v2.0/token\", \"UserRealmEndpoint\":\"https://{host}/common/UserRealm\"}";
```

Comment:
I am not sure what is this PR attempting to fix. This library does not support v2 endpoint. for v2 needs please use MSAL.